### PR TITLE
Add question 2a to transit visa branch to track Common Travel Areas (v2)

### PR DESCRIPTION
**Pull request by floehopper**
Supersedes #2553. Opened on behalf of @ikennaokpala, because the branch name in the previous version was causing the following CI builds to fail:

* https://ci.dev.publishing.service.gov.uk/job/govuk_smartanswers_branches/5238/console
* https://ci.dev.publishing.service.gov.uk/job/govuk_smart_answers_regressions/3905/console

[Trello Card](https://trello.com/c/LrqmkiU5/149-check-if-you-need-a-uk-visa-new-transit-visa-logic-and-content)

## Motivation

The current state of check uk visa smart-answer doesn't support or provide information to users who are coming to the UK on transit to the Channel Islands, Isle of Man or Republic of Ireland (known collectively as the ‘common travel area’ or ‘CTA’).

Upon request from the Content Team ([via this logic document](https://docs.google.com/document/d/1HRO23JljpHboQOzFQ6gSc2ks7fRfpDaXjEEpR0iOkx4/edit)), I have added a new question (2a) and outcome (14a) to the transit branch of this smart answer.

I have changed the content of Question 3. According to the Content team, the current definition for users passing through UK border control isn’t quite right.

## Factcheck New Question 2a

[Preview Link](https://smart-answers-pr-2553.herokuapp.com/check-uk-visa/y/syria/transit)
[GOV.UK](https://gov.uk/check-uk-visa/y/syria/transit/)

### Expected changes
* Added new Question (2a)

### Before

screen shot 2016-05-27 at 16 55 01

### After

screen shot 2016-05-27 at 16 55 09

## Factcheck New Outcome 14a

[Preview Link](https://smart-answers-pr-2553.herokuapp.com/check-uk-visa/y/syria/transit/yes)
[GOV.UK](https://gov.uk/check-uk-visa/y/syria/transit/yes)

### Expected changes
* Added new outcome (14a)

### Before

screen shot 2016-05-27 at 16 56 34

### After

screen shot 2016-05-27 at 16 56 43

## Factcheck Question 3

[Preview Link](https://smart-answers-pr-2553.herokuapp.com/check-uk-visa/y/syria/transit/no)
[GOV.UK](https://gov.uk/check-uk-visa/y/syria/transit/no)

### Expected changes
* Content changes to Question 3

### Before

screen shot 2016-05-27 at 16 59 23

### After

screen shot 2016-05-27 at 16 59 31

### CODE REVIEW

**Comment by floehopper** [comment]
I've added a fixup commit to fix the Rubocop violation, because I wanted to see the CI build pass. You can incorporate that commit using the following commands:

```
$ git co check-uk-visa-changes-to-transit-visa-for-cta
$ git rebase --interactive --autosquash master
```

**Comment by ikennaokpala** [comment]
@floehopper thanks for this.. it weird that branch name made the build to fail.

**Comment by chrisroos** [comment]
@ikennaokpala / @floehopper: Can this be reviewed given our recent change in workflow?

**Comment by ikennaokpala** [comment]
@chrisroos This isn't flowing the new workflow.. will let everyone know when it is available for review

**Comment by floehopper** [comment]
@ikennaokpala: I know you've said that you don't want this reviewed at the moment, but I thought I should give you a heads-up that at first glance the first commit in this PR looks like it contains a load of different changes. Similarly the commit which updates the artefacts contains a lot of changes with no explanation in the commit note. Thus I suspect it's going to be hard to review. This relates to what we were talking about in the recent retrospective. Anyway, I thought it was worth mentioning.

**Comment by ikennaokpala** [comment]
@floehopper noted.. I will review and make the amends

**Comment by floehopper** [comment]
Minor: There are very long lines in most of the commit notes. I'd really recommend configuring your editor to avoid this happening.

File: test/integration/smart_answer_flows/check_uk_visa_test.rb [hunk 1]
```diff
@@ -7,11 +7,42 @@ class CheckUkVisaTest < ActiveSupport::TestCase
   include FlowTestHelper
 
   setup do
-    @location_slugs = %w(andorra anguilla armenia bolivia canada china colombia croatia mexico south-africa stateless-or-refugee syria turkey yemen oman united-arab-emirates qatar taiwan venezuela)
+    @location_slugs = %w(afghanistan andorra anguilla armenia bolivia canada china colombia croatia mexico south-africa stateless-or-refugee syria turkey yemen oman united-arab-emirates qatar taiwan venezuela)
```
Comment (floehopper) [inline test/integration/smart_answer_flows/check_uk_visa_test.rb]: Minor: Unless it's critical, I'd be tempted to use one of the countries already in this list in your new tests, so you don't have to add `afghanistan` to the list.

File: lib/smart_answer_flows/check-uk-visa/outcomes/outcome_transit_to_the_cta.govspeak.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<% content_for :title do %>
+  You’ll need a visa to pass through the UK (unless you’re exempt)
+<% end %>
+
+<% content_for :body do %>
+
```
Comment (floehopper) [inline lib/smart_answer_flows/check-uk-visa/outcomes/outcome_transit_to_the_cta.govspeak.erb]: Very minor: I don't think we need this blank line or the one at the bottom of the body.

**Comment by floehopper** [comment]
Minor: When adding/removing nodes to a flow and/or changing the routing of a flow, it can be helpful to include a screenshot of the appropriate bit of the flow visualisation before and after the change.

**Comment by floehopper** [comment]
Well done for splitting up the original first commit into two commits - that's helpful.

**Comment by floehopper** [comment]
Most of my comments are pretty minor, but it would be nice to at least make the changes to the commit notes that I've suggested, since that shouldn't be too much work.

In the bigger scheme of things, the only concern I have is that the changes to the regression test artefacts in the 1st "Regenerate regression test artefacts (check-uk-visa)" commit (7bf63e34507845d815b50fd8dad45b42d2cdcb36) are not explained in the commit note and I haven't checked through them in detail to be sure that they are as we would expect.

**Comment by floehopper** [comment]
For the record, I just chatted to @pmanrubia & @ikennaokpala about the 1st "Regenerate regression test artefacts (check-uk-visa)" commit (7bf63e3) and we've agreed that in this case it's not worth explaining in details why the changes are as we expect, but @ikennaokpala is going to add something to the commit note to explain that we're happy that the changes are as we expect, because of the new integration tests and because of extensive manual testing.

File: test/integration/smart_answer_flows/check_uk_visa_test.rb (hunk 1 above)
Comment (ikennaokpala) [inline test/integration/smart_answer_flows/check_uk_visa_test.rb]: In the commit note I have mentioned why I added afghanistan

**Comment by ikennaokpala** [comment]
@floehopper Thanks for your input and guidance on this new approach to adding nodes ina smart answer. I will be closing this PR to create a new one.. According to the team, we expect there is still more to be done as the current solution (that introduces the new cta node) isn't sufficient.

**Pull request closed without merging**